src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,219 @@
+import React, { useState, useEffect, useRef } from 'react';
+import classNames from 'classnames';
+import { Error } from '../../types/Error';
+import { USER_ID, deleteTodo, updateTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  id: number;
+  title: string;
+  completed: boolean;
+  setErrorMessage: React.Dispatch<React.SetStateAction<Error>>;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  isInputLoading?: boolean;
+  deletingTodoIds: number[];
+};
+
+export const TodoItem: React.FC<Props> = ({
+  id,
+  title,
+  completed,
+  setErrorMessage,
+  setTodos,
+  isInputLoading = false,
+  deletingTodoIds,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [isCompleted, setIsCompleted] = useState(completed);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedTodoTitle, setEditedTodoTitle] = useState(title);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isEditing) {
+      inputRef.current?.focus();
+    }
+  }, [isEditing]);
+
+  useEffect(() => {
+    setIsCompleted(completed);
+    setEditedTodoTitle(title);
+  }, [completed, title]);
+
+  function handleDeleteTodo(todoId: number) {
```

Comment:
It is important to stick to one way of functions, fix it everywhere
```suggestion
  const handleDeleteTodo = (todoId: number) => {
```